pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1534,6 +1538,12 @@ def _handle_distribution_tree(declarative_content):
                 for update_reference in update_references:
                     update_reference.update_record = update_record
                     update_references_to_save.append(update_reference)
+            elif isinstance(declarative_content.content, Package):
+                artifact = declarative_content.d_artifacts[0]  # Packages have 1 artifact
+                signer_key_id = parse_signer_id(artifact.artifact.file.path)
```

Comment:
I don't think `path` will work when Pulp's artifact storage is S3 / Azure.  We could use `.read()` and save it into a temporary file, then parse that, but it would not be terribly efficient.  But short term it might be the only way... what we need is a way to parse the downloaded files before they get shipped off from temporary storage to the storage backend.

That is too invasive of a change to handle in this PR though.

The question is whether we want to live with that inefficiency or disable the feature if the user is using a cloud storage backend.